avionics/envs/x86/lib/HAL/time.cpp

```diff
@@ -1,31 +1,26 @@
+#include "stdio_controller.hpp"
 #include <HAL/time.h>
 
 namespace Hal {
-/**
- * @brief Sleep for the given std::chrono::duration. Only available for x86 -
- * Arduino framework doesn't implement enough of <chrono>.
- * @param t Duration to sleep for.
- */
-template <class Duration> void sleep(Duration t) {
-    namespace c = std::chrono; // alias
-    c::steady_clock::time_point start = c::steady_clock::now();
-    c::steady_clock::time_point tgt =
-        start + t; // Precalculate so checks can happen more often?
-    while (c::steady_clock::now() < tgt)
-        ;
-}
-
 /**
  * @brief Sleep for some number of milliseconds.
  * @param t Duration to sleep for, in milliseconds.
  */
-void sleep_ms(uint32_t t) { sleep(ms(t)); }
+void sleep_ms(uint32_t t) {
+    if (t >= (std::numeric_limits<uint32_t>::max() / 1000))
+        throw "The given sleep length will cause overflow.";
```

Comment:
Can't say I know too much about C++ exceptions, but my understanding is that you should throw things related to std::exception, and I'm not sure how string literals will behave with stack unwinding and potential undefined behaviour - see https://stackoverflow.com/questions/27179011/c-exception-throw-a-string and https://stackoverflow.com/questions/27179011/c-exception-throw-a-string